enhancements/network/ocp-on-smart-nics.md

```diff
@@ -0,0 +1,276 @@
+---
+title: Installing OCP on ARM-Based Smart NICs
+authors:
+  - "@danwinship"
+reviewers:
+  - "@russellb"
+  - "@dcbw"
+  - "@zshi-redhat"
+approvers:
+  - "@russellb"
+  - "@dcbw"
+creation-date: 2021-04-15
+last-updated: 2021-04-15
+status: provisional
+---
+
+# Installing OCP on ARM-Based Smart NICs
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+As servers have greater and greater networking needs, hardware vendors
+have created increasingly more complex and powerful network cards. The
+latest iteration of this is "Smart NICs" that include an entire ARM
+system running Linux on the NIC. With such a NIC, we can offload
+nearly all network processing from the host computer onto the NIC
+"DPU" (Data Processing Unit), freeing up host CPU resources for
+customer workloads (and allowing faster data transfer by ensuring the
+network stack doesn't need to compete for CPU with the customer
+workload).
+
+However, doing this requires installing and managing an appropriate
+software stack on all of the NICs in the cluster. This software will
+need to be kept in sync with the software on the host PC (eg, as new
+functionality is added to OCP). The obvious answer to this problem is
+to run OpenShift on the NIC as well as on the host.
+
+The initial target of this work is the [NVIDIA BlueField-2 DPU].
+However, similar products are being worked on by other vendors which
+we will eventually want to support, so the solution should not be too
+specialized to this one product.
+
+[NVIDIA BlueField-2 DPU]: https://www.nvidia.com/en-us/networking/products/data-processing-unit/
+
+## Motivation
+
+### Goals
+
+- Extend OCP as needed to support installation on Smart NICs such as
+  the BlueField-2 where:
+
+    1. The NIC is capable of running OpenShift.
+
+    2. The NIC has an IPMI-addressable Baseboard Management Controller
+       that can be used to control and provision it independently of
+       its host.
+
+    3. The NIC implements SR-IOV in such a way that we can connect
+       pods on the host directly to an OVN bridge on the NIC.
```

Comment:
Worth a blurb about third-party CNIs other than ovn-kubernetes.. even if that blurb is "they're on their own."